src/CarbonAware.WebApi/test/unitTests/Configuration/ServiceCollectionExtensionsTests.cs

```diff
@@ -0,0 +1,106 @@
+using CarbonAware.WebApi.Configuration;
+using NUnit.Framework;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using NUnit.Framework.Internal;
+
+namespace CarbonAware.WepApi.UnitTests;
+
+[TestFixture]
+public class ServiceCollectionExtensionsTests
+{
+    [Test]
+    public void AddMonitoringAndTelemetry_AddsServices_WithConnectionString()
+    {
+        // Arrange
+        IServiceCollection services = new ServiceCollection();
+
+        var inMemorySettings = new Dictionary<string, string>
+        {
+            { "CarbonAwareVars:TelemetryProvider", "ApplicationInsights" },
+            { "ApplicationInsights_Connection_String", "AppInsightsConnectionString" }
+        };
+        IConfiguration configuration = new ConfigurationBuilder()
+            .AddInMemoryCollection(inMemorySettings)
+            .Build();
+
+        // Act & Assert
+        Assert.DoesNotThrow(() => ServiceCollectionExtensions.AddMonitoringAndTelemetry(services, configuration));
+        Assert.AreEqual(services.Count(), 43);
```

Comment:
QQ: How do you know this exact number?

Reply:
I ran `AddMonitoringAndTelemetry` and checked the number of services registered to the collection when it runs successfully. I added it to contrast that when this is not successful, the number of services registered is `0` in the other tests.

Thinking about it in depth, we don't need to add how many services the collection has here as long as the other tests demonstrate the number is `0` when the call throws an error (also this `43` may change in the future lol). I removed this assertion statement and the one below, let me know if that looks good!